parlai/mturk/tasks/acute_eval/analysis.py

```diff
@@ -308,19 +348,46 @@ def get_max_hits_per_worker(self) -> List[int]:
         """
         return self.dataframe.groupby('worker')['run_id'].count().max()
 
+    def filter_by_dialogue_length(self, is_debug=False):
+        """
+        Filter out matchup with one of the conversation shorter than
+        self.min_dialogue_length This applies to calculating sorted_win_frac_df and
+        signficance_df, but not html visualizations of conversations.
+
+        :param is_debug: if True, print logs indicating the number of pairings filtered out due to short conversation.
+            is_debug bool
+        """
+        df = pd.DataFrame()
+        filter_list = {}
+        for _, row in self.dataframe.iterrows():
+            keep_row = True
+            for model_name, dialogue_length in row['dialogue_lengths'].items():
+                if keep_row and dialogue_length < self.min_dialogue_length:
+                    keep_row = False
+                    filter_list[model_name] = filter_list.get(model_name, 0) + 1
+            if keep_row:
+                df = df.append(row, ignore_index=True)
+        if is_debug:
+            for model_name in filter_list:
+                print(
+                    f"For {self.run_id}: filter out {filter_list[model_name]} matchups due to {model_name} with dialogue length shorter than {self.min_dialogue_length}"
+                )
+        return df
+
     def get_wins_per_model_matchup(self) -> pd.DataFrame:
         """
         Return the wins for each model by matchup.
         """
+        df_filtered = self.filter_by_dialogue_length(True)
```

Comment:
if we're just calling this with `True` I say we drop the function argument and always print that message above

Reply:
Good question! My point is that, if missing this argument, the same message of 'filter out XXX' will get printed twice as the filter function gets called in both get_matchup_totals_with_signficance and get_wins_per_model_matchup, which I think it's kind of redundant so.

Reply:
oh great, i missed that second call, let's keep it then!